### PR TITLE
Add proto message for pruning acks, msgs, stubs for keeper/msg-server.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #5410

### Commit Message / Changelog Entry

```text
feat: add proto message for pruning acks, 
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: proto/ibc/core/channel/v1/tx.proto [hunk 1]
```diff
@@ -444,3 +447,17 @@ message MsgUpdateParams {
...
 message MsgUpdateParamsResponse {}
+
+// MsgPruneAcknowledgements defines the request type for the PruneAcknowledgements rpc.
+message MsgPruneAcknowledgements {
+  option (cosmos.msg.v1.signer)      = "signer";
+  option (gogoproto.goproto_getters) = false;
+
+  string port_id      = 1;
+  string channel_id   = 2;
+  uint64 num_to_prune = 3;
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply proto/ibc/core/channel/v1/tx.proto:458]: Maybe `limit`?

File: modules/core/04-channel/types/msgs_test.go [hunk 2]
```diff
@@ -1188,3 +1190,77 @@ func (suite *TypesTestSuite) TestMsgChannelUpgradeCancelGetSigners() {
...
+			host.ErrInvalidID,
+		},
+		{
+			"invalid channel identifier",
+			func() {
+				msg.ChannelId = invalidChannel
+			},
+			types.ErrInvalidChannelIdentifier,
+		},
+
```
Comment (crodriguezvega) [inline modules/core/04-channel/types/msgs_test.go:1228]: Empty line!

```suggestion

```

**Review by crodriguezvega — APPROVED** [review]
Nice!

File: proto/ibc/core/channel/v1/tx.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (colin-axner) [reply proto/ibc/core/channel/v1/tx.proto:458]: I like limit

**Review by colin-axner — APPROVED** [review]
Thanks for splitting up these changes. Really easy to review! LGTM!!

File: proto/ibc/core/channel/v1/tx.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (colin-axner) [reply proto/ibc/core/channel/v1/tx.proto:458]: me too haha

File: modules/core/04-channel/types/errors.go [hunk 3]
```diff
@@ -51,4 +51,5 @@ var (
 	ErrInvalidUpgradeTimeout           = errorsmod.Register(SubModuleName, 35, "upgrade timeout is invalid")
 	ErrPendingInflightPackets          = errorsmod.Register(SubModuleName, 36, "pending inflight packets exist")
 	ErrUpgradeTimeoutFailed            = errorsmod.Register(SubModuleName, 37, "upgrade timeout failed")
+	ErrInvalidPruningAmount            = errorsmod.Register(SubModuleName, 38, "invalid pruning amount")
```
Comment (damiannolan) [inline modules/core/04-channel/types/errors.go:54]: I guess `ErrInvalidPruningLimit` or something like that if we're going with `limit`

File: modules/core/04-channel/types/msgs_test.go [hunk 4]
```diff
@@ -1188,3 +1190,77 @@ func (suite *TypesTestSuite) TestMsgChannelUpgradeCancelGetSigners() {
...
+		{
+			"missing signer address",
```
Comment (damiannolan) [inline modules/core/04-channel/types/msgs_test.go:1230]:
```suggestion
			"empty signer address",
```

**Review by damiannolan — APPROVED** [review]
Awesome work! 🚀 

I'm totally on board with `limit` as the naming choice!

**Pull request merged**